acme/api/middleware.go

```diff
@@ -422,11 +423,20 @@ func verifyAndExtractJWSPayload(next nextHTTP) nextHTTP {
 			render.Error(w, acme.NewError(acme.ErrorMalformedType, "verifier and signature algorithm do not match"))
 			return
 		}
+
 		payload, err := jws.Verify(jwk)
-		if err != nil {
+		switch {
+		case errors.Is(err, jose.ErrCryptoFailure):
+			payload, err = retryVerificationWithPatchedSignatures(jws, jwk)
+			if err != nil {
+				render.Error(w, acme.WrapError(acme.ErrorMalformedType, err, "error verifying jws with patched signature(s)"))
+				return
+			}
```

Comment:
Yeah, I think with my latest change it should be safe to always check/patch it, and it was how I implemented it in the very first iteration. But I think it's a little bit clearer what happens doing it like this (with a distinct error), than to always first check and patch if needed. It also should only be the case for (current) Apple devices, and I think it's a bit nicer to not have this sit in a hot code path. Besides that, it _should_ be a temporary solution, so it can be removed easily.